### PR TITLE
Register fastboot transform

### PR DESCRIPTION
**Pull request by kratiahuja**
This PR uses the new ember-cli API specified in RFC: https://github.com/ember-cli/rfcs/pull/108

Tested with a dummy app and npm-linking.

Fixes issues https://github.com/ember-fastboot/ember-cli-fastboot/issues/413, https://github.com/ember-fastboot/ember-cli-fastboot/issues/433 etc

**NOTE**: This will ease importing fastboot incompatible libraries from app and NOT addon. For addon, I will put up a PR in `fastboot-transform` to provide the necessary API.

- [x] Upgrade addon to ember-cli 2.16
- [x] Add tests

# TODO
- [ ] Update README
- [ ] Update fastboot-website

### CODE REVIEW

**Comment by izelnakri** [comment]
ember-cli v2.16 released today. Can we merge this?

**Comment by kratiahuja** [comment]
There are still TODO items. I will work on it during the weekend.

**Comment by kratiahuja** [comment]
Tests are failing due to unrelated errors. will debug them next week.

**Comment by fastindian84** [comment]
Is there any chance, that this feature will be merged?

**Comment by kratiahuja** [comment]
There are same failing tests that I haven't got a chance to look at. If someone can give a hand, would be nice.

**Comment by kratiahuja** [comment]
OK fixed all failing tests. THis should be good to review. @rwjblue whenever you have a chance, could you please review this?

File: package.json [hunk 1]
```diff
@@ -59,6 +59,8 @@
     "ember-resolver": "^4.3.0",
     "ember-sinon": "^1.0.0",
     "ember-source": "~2.16.0",
+    "fastboot-transform": "^0.1.2",
+    "fs-extra": "^4.0.0",
```
Comment (rwjblue) [inline package.json]: Why do we need this? I don’t see it being used here in this PR?

Comment (kratiahuja) [reply package.json]: Sorry this was a merge conflict issue.

**Comment by kratiahuja** [comment]
@rwjblue updated. could you please take another look?

File: package.json [hunk 2]
```diff
@@ -59,6 +58,8 @@
     "ember-resolver": "^4.3.0",
     "ember-sinon": "^1.0.0",
     "ember-source": "~2.16.0",
+    "fastboot-transform": "^0.1.2",
```
Comment (SergeAstapov) [inline package.json]: @kratiahuja shouldn't `fastboot-transform` be listed in `dependencies`? If I understand correctly it needs to be available for the application at build time.

Comment (rwjblue) [reply package.json]: Yes,confirm

Comment (SergeAstapov) [reply package.json]: Thank you @rwjblue for confirmation, I’ll make PR to fix this

**Pull request merged**